### PR TITLE
Update JSDoc type and param info for Page/Site.js

### PR DESCRIPTION
**Pull request by le0tan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, please explain:

Related to #1027 

**What changes did you make? (Give an overview)**

* Add typedefs for configuration objects in `Page.js`.
* Add property typedefs for Page object.
* Updated some outdated JSDoc params in both `Site.js` and `Page.js`.

**Provide some example code that this change will affect:**

na

**Is there anything you'd like reviewers to focus on?**

Whether the documentation is accurate.

**Testing instructions:**

No functional changes. No testing needed.

**Proposed commit message: (wrap lines at 72 characters)**

TODO

### CODE REVIEW

**Comment by le0tan** [comment]
@alyip98 @nbriannl thanks for the input! I will request another review after settling some of the comments.

> Another PR can be made to flesh out the documentation even further, which will require someone who understands what's going on in Site and Page.

I think @alyip98 showed us some documentation on this before?

File: src/Page.js [hunk 1]
```diff
@@ -55,48 +55,212 @@ cheerio.prototype.options.xmlMode = true; // Enable xml mode for self-closing ta
...
+   * @property {Set<any>} baseUrlMap
+   * @property {string} content
+   * @property {string} faviconUrl
+   * @property {Object<string, any>} frontmatter
+   * @property {string} layout
+   * @property {string} layoutsAssetPath
+   * @property {string} rootPath
+   * @property {boolean} enableSearch
+   * @property {boolean} globalOverride
+   * @property {Array} plugins
```
Comment (marvinchin) [inline src/Page.js:72]: Should we label the type of the content in the array?

File: src/Page.js [hunk 2]
```diff
@@ -847,8 +1027,21 @@ class Page {
...
 
+  /**
+   * A plugin configuration object.
+   * @typedef {Object<string, any>} PluginConfig
+   * @property {number} headingIndexingLevel
+   * @property {boolean} enableSearch
+   * @property {boolean} searchable
+   * @property {string} rootPath
+   * @property {string} sourcePath
+   * @property {Set} includedFiles
```
Comment (marvinchin) [inline src/Page.js:1038]: Same here, should we label the contents of the set?

**Comment by alyip98** [comment]
> I think @alyip98 showed us some documentation on this before?

It's rather outdated and I don't have plans to update it, so feel free to take over if you'd like 
[link](https://github.com/alyip98/markbind/blob/812-documentation/docs/devGuide/activity.md)
[pdf](https://github.com/MarkBind/markbind/files/4432139/Markbind.Page.and.Site.Reference.pdf)

File: src/Page.js (hunk 1 above)
Comment (le0tan) [reply src/Page.js:72]: The content in `plugin` array is the `require`d plugin objects (e.g. `plugins['name']=require('name');`)

**Pull request merged**